src/result.ts

```diff
@@ -62,7 +62,11 @@ export type Result<T, E> = Ok<T, E> | Err<T, E>
 
 export const ok = <T, E = never>(value: T): Ok<T, E> => new Ok(value)
 
-export const err = <T = never, E = unknown>(err: E): Err<T, E> => new Err(err)
+export function err<T = never, E extends string = string>(err: E): Err<T, E>
```

Comment:
Can do this with an overloaded type if you'd prefer, but figured this was the easiest way to make this change.